api/v1alpha1/common.proto

```diff
@@ -0,0 +1,76 @@
+import "gogoproto/gogo.proto";
+import "google/api/field_behavior.proto";
+import "k8s.io/api/core/v1/generated.proto";
+
+package istio_operator.v2.api.v1alpha1;
+
+option go_package = "github.com/banzaicloud/istio-operator/v2/api/v1alpha1";
+
+message NamespacedName {
+    // Name of the referenced Kubernetes resource
+    string name = 1 [(gogoproto.jsontag) = "name,omitempty"];
+    // Namespace of the referenced Kubernetes resource
+    string namespace = 2  [(gogoproto.jsontag) = "namespace,omitempty"];;
+}
+
+message BaseK8sResourceConfigurationWithHPAWithoutImage {
+    // Standard Kubernetes replica count configuration
+    // +kubebuilder:validation:Minimum=0
+    google.protobuf.Int32Value ReplicaCount = 1 [(gogoproto.wktpointer) = true, (gogoproto.jsontag) = "replicaCount,omitempty"];
+    // Standard Kubernetes minimum replicas configuration
+	// +kubebuilder:validation:Minimum=0
+    google.protobuf.Int32Value MinReplicas = 2 [(gogoproto.wktpointer) = true, (gogoproto.jsontag) = "minReplicas,omitempty"];
+    // Standard Kubernetes maximum replicas configuration
+	// +kubebuilder:validation:Minimum=0
+    google.protobuf.Int32Value MaxReplicas = 3 [(gogoproto.wktpointer) = true, (gogoproto.jsontag) = "maxReplicas,omitempty"];
+    // Standard Kubernetes resource configuration
+    BaseK8sResourceConfiguration BaseK8sResourceConfig = 4 [(gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = ",inline"];
+}
+
+message BaseK8sResourceConfiguration {
+    // Standard Kubernetes memory and CPU resource requirements
+    k8s.io.api.core.v1.ResourceRequirements Resources = 1 [(gogoproto.jsontag) = "resources,omitempty"];
+    // Standard Kubernetes node selector configuration
+	map<string,string> NodeSelector = 2 [(gogoproto.jsontag) = "nodeSelector,omitempty"];
+    // Standard Kubernetes affinity configuration
+	k8s.io.api.core.v1.Affinity Affinity = 3 [(gogoproto.jsontag) = "affinity,omitempty"];
+    // Standard Kubernetes toleration configuration
+	repeated k8s.io.api.core.v1.Toleration Tolerations = 4 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "tolerations,omitempty"];
+    // Standard Kubernetes pod annotation configuration
+	map<string,string> PodAnnotations = 5 [(gogoproto.jsontag) = "podAnnotations,omitempty"];
+    // Standard Kubernetes security context configuration
+	k8s.io.api.core.v1.SecurityContext SecurityContext = 6 [(gogoproto.jsontag) = "securityContext,omitempty"];
+}
+
+message GatewaySDSConfiguration {
+    // Flags if Gateway Secret Discovery Service is enabled or not
+    google.protobuf.BoolValue Enabled = 1 [(gogoproto.wktpointer) = true, (gogoproto.jsontag) = "enabled,omitempty"];
+    // Container image to use for SDS
+    string Image = 2 [(gogoproto.jsontag) = "image,omitempty"];
+    // Standard Kubernetes memory and CPU resource requirements
+    k8s.io.api.core.v1.ResourceRequirements Resources = 3 [(gogoproto.jsontag) = "resources,omitempty"];
+}
```

Comment:
I might remember incorrectly, but aren't these settings for the `node-agent` component, which was deprecated? If so, then we probably won't need these settings at all. @waynz0r, could you help out in this and confirm or deny what I wrote? :)

(Also, even if we should keep this config, imho this should go to the meshgateway spec instead of the common spec.)

Reply:
you're probably right, I was blindly copying fields from the previous CR and missed it. I'll delete it.